### ISSUE TITLE
Remove deprecated fast-sync parameters

### ISSUE THREAD

**Issue by williambanfield** [tendermint/tendermint#6908]
During the `0.35` release cycle, the `fast-sync` command line and configuration parameters were renamed to reflect the changed package name from `fastsync` to `blocksync`. Reference to these parameters was left in the codebase to prompt users to update their configuration if the paramters were encountered. 

By 0.36, users will have removed these parameters and updated to the newer parameter names. The checks that were added in the 0.35 release cycle should be removed so that references to the old parameters are removed entirely.

These parameters were renamed in #6896 .

**Comment by marbar3778** [comment]
I propose we remove the option of turning blocksync off in general. I cant fathom a reason someone would want to sync via the sluggish consensus pkg.

**Comment by cmwaters** [comment]
Yeah I think when we add fallback to blocksync logic when consensus starts to lag, we should remove the ability to turn blocksync off and just have it as a permanent fixture.

**Comment by marbar3778** [comment]
why wait till then? seems like doing it now opposed to doing it later, there isnt really a difference other than no we have less params.
